<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetNoConstraint</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="groupCamera">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Camera setup</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="toolTip">
            <string>Horizontal field of view of the final image in degress. Minimum 0, maximum 170.</string>
           </property>
           <property name="text">
            <string>Field of view</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineFov">
           <property name="text">
            <string>60</string>
           </property>
           <property name="placeholderText">
            <string>Horizontal FOV</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="toolTip">
            <string>Size in pixels of the final image. Positive, non-zero integer.</string>
           </property>
           <property name="text">
            <string>Image size</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="toolTip">
            <string>Position in 3D of the camera, up is (0,1,0). Double.</string>
           </property>
           <property name="text">
            <string>Camera position</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_5">
           <property name="toolTip">
            <string>Direction in which the camera is looking, up is (0,1,0). Double.</string>
           </property>
           <property name="text">
            <string>Camera Orientation</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_6">
           <property name="toolTip">
            <string>Number of rays thrown per pixels, more rays means accuracy but longer computational time. Positive, non-zero integer.</string>
           </property>
           <property name="text">
            <string>Sampling rate</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLineEdit" name="lineWidth">
             <property name="text">
              <string>600</string>
             </property>
             <property name="placeholderText">
              <string>Width in pixels</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineHeight">
             <property name="text">
              <string>450</string>
             </property>
             <property name="placeholderText">
              <string>Height in pixels</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLineEdit" name="lineCamX">
             <property name="text">
              <string>0</string>
             </property>
             <property name="placeholderText">
              <string>x coordinate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineCamY">
             <property name="text">
              <string>0</string>
             </property>
             <property name="placeholderText">
              <string>y coordinate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineCamZ">
             <property name="text">
              <string>0</string>
             </property>
             <property name="placeholderText">
              <string>z coordinate</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLineEdit" name="lineLookingX">
             <property name="text">
              <string>0</string>
             </property>
             <property name="placeholderText">
              <string>x coordinate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineLookingY">
             <property name="text">
              <string>0</string>
             </property>
             <property name="placeholderText">
              <string>y coordinate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineLookingZ">
             <property name="text">
              <string>-1</string>
             </property>
             <property name="placeholderText">
              <string>z coordinate</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="lineSamplingRate">
           <property name="text">
            <string>100</string>
           </property>
           <property name="placeholderText">
            <string>Rays per pixel</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_7">
           <property name="toolTip">
            <string>Macimum number of bounces each ray can do, recommended is 5. Positive, non-zero integer.</string>
           </property>
           <property name="text">
            <string>Max bounces</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="lineDepth">
           <property name="toolTip">
            <string/>
           </property>
           <property name="text">
            <string>5</string>
           </property>
           <property name="placeholderText">
            <string>Depth of recursion</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item alignment="Qt::AlignBottom">
       <widget class="QLabel" name="label">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>List of the objects in the scene :</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="listWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>192</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="buttonAddObject">
          <property name="text">
           <string>Add Object</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonUpdateList">
          <property name="text">
           <string>Update List</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLineEdit" name="lineJsonFileName">
          <property name="placeholderText">
           <string>Enter file name to save/load the list</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonLoadJson">
          <property name="text">
           <string>Load Obj. list</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonSaveJson">
          <property name="text">
           <string>Save Obj. list</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item alignment="Qt::AlignBottom">
         <widget class="QPushButton" name="buttonRender">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Render</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignBottom">
         <widget class="QLineEdit" name="lineEdit">
          <property name="placeholderText">
           <string>Enter file name for the image.</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignBottom">
         <widget class="QPushButton" name="buttonLoad">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Load Image</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignBottom">
         <widget class="QPushButton" name="buttonSaveImage">
          <property name="text">
           <string>Save Image</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>843</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
